### PR TITLE
Dropping usage of _Monkey in favor of mock.patch.

### PR DESCRIPTION
**Pull request by dhermes**
This was done only in bigquery, datastore and storage packages. Still needs updates in bigtable, core, logging, monitoring, pubsub and speech.

From discussion with @tseaver in #2681.

### CODE REVIEW

File: datastore/unit_tests/test__http.py [hunk 1]
```diff
@@ -417,16 +420,18 @@ def mock_api(connection):
             connections.append(connection)
             return return_val
 
-        with _Monkey(MUT, _DatastoreAPIOverHttp=mock_api):
+        patch = mock.patch(
```
Comment (daspecster) [inline datastore/unit_tests/test__http.py]: Why not...
```python
method_name = 'google.cloud.datastore._http._DatastoreAPIOverHttp'
with mock.patch(method_name, new=mock_api):
```

Comment (dhermes) [inline datastore/unit_tests/test__http.py]: @daspecster That's fine with me. I have no preference here.

Comment (daspecster) [inline datastore/unit_tests/test__http.py]: Ok, I was mostly curious if you had a reason for expanding it out.

**Comment by dhermes** [comment]
@tseaver Any reservations?

File: datastore/unit_tests/test__http.py (hunk 1 above)
Comment (dhermes) [inline datastore/unit_tests/test__http.py]: Nope no reason.

**Comment by tseaver** [comment]
@dhermes LGTM after rebase, Travis OK.

**Pull request merged**